psl-core/src/main/java/org/linqs/psl/application/learning/weight/bayesian/GaussianProcessPrior.java

```diff
@@ -217,53 +263,89 @@ private void setWeights(WeightConfig config) {
+        }
+
         // If systematic generation of points will lead to not a reasonable exploration of space,
         // then just pick random points in space and hope it is better than being systematic.
 
         if (randomConfigsOnly) {
             log.debug("Generating random configs.");
-            return getRandomConfigs();
-        }
-
-        if (numPerSplit < 5) {
-            log.warn("Note not picking random points and large number of rules will yield bad exploration.");
-        }
+            configs = getRandomConfigs();
+            if (useProvidedWeight) {
+                configs.add(PROVIDED_CONFIG_INDEX, initialConfig);
+            }
+        } else {
 
-        float inc = max / numPerSplit;
-        float[] configArray = new float[numMutableRules];
-        Arrays.fill(configArray, min);
-        WeightConfig config = new WeightConfig(configArray);
-
-        boolean done = false;
-        while (!done) {
-            int i = 0;
-            configs.add(new WeightConfig(config));
-            for (int j = 0; j < numMutableRules; j++) {
-                if (config.config[i] < max) {
-                    config.config[i] += inc;
-                    break;
-                }
+            if (numPerSplit < 5) {
+                log.warn("Note not picking random points and large number of rules will yield bad exploration.");
+            }
 
-                if (i == numMutableRules - 1) {
-                    done = true;
-                    break;
+            float inc = max / numPerSplit;
+            float[] configArray = new float[numMutableRules];
+            Arrays.fill(configArray, min);
+            WeightConfig config = new WeightConfig(configArray);
+
+            boolean done = false;
+            while (!done) {
+                int i = 0;
+                configs.add(new WeightConfig(config));
+                for (int j = 0; j < numMutableRules; j++) {
+                    if (config.config[i] < max) {
+                        config.config[i] += inc;
+                        break;
+                    }
+
+                    if (i == numMutableRules - 1) {
+                        done = true;
+                        break;
+                    }
+
+                    config.config[i] = min;
+                    i++;
                 }
-
-                config.config[i] = min;
-                i++;
             }
         }
-
         return configs;
     }
 
     private List<WeightConfig> getRandomConfigs() {
         int numMutableRules = this.mutableRules.size();
         List<WeightConfig> configs = new ArrayList<WeightConfig>();
+
+        if (searchHypersphere) {
+            log.debug("Generating random points from hypersphere");
+        } else if (searchDirichlet) {
+            log.debug("Generating random points from dirichlet");
+        } else {
+            log.debug("Generating random points from hypercube");
+        }
+
         for (int i = 0; i < maxConfigs; i++) {
```

Comment:
Instead if re-implementing this code in multiple places, can we make a single WeightSampler class?
Then all the search-based methods that sample weights can easily use it?
Also, that new class can have a single set of options that controls how weights are sampled.

Reply:
The WeightSampler was added to the org.linqs.psl.application.learning.weight.search package and implements this  functionality.

Reply:
Does that mean this comment is resolved?
If so, mark it as resolved.
(Also, only make comments when you want the conversation to continue (otherwise just use the resolve button). I get emails for every comment.)